### PR TITLE
Convert benefits to a list.

### LINKED ISSUE #964: Turn client data current benefits into a list/array of benefit names
To be most flexible for adding future benefits, it'd work better to build a list of what benefits the person has.

### PR DESCRIPTION
**Pull request by turnerhayes**
Fixes #964

### CODE REVIEW

File: src/forms/output/BenefitsLineGraph.js [hunk 1]
```diff
@@ -48,8 +48,12 @@ class BenefitsLineGraph extends Component {
         interval  = Math.ceil((max / 100) / 10) * 10;
 
     var xRange      = _.range(limits.min, max, interval),  // x-axis/earned income numbers
-        extraProps  = { snap: { fill: false }, section8: { fill: false }},
+        extraProps  = {},
         datasets    = getChartData(xRange, multiplier, client, activePrograms, extraProps);
+    
+    for (let benefitIndex = 0; benefitIndex < activePrograms.length; benefitIndex++) {
+      extraProps[ activePrograms[ benefitIndex ] ] = { fill: false };
```
Comment (knod) [inline src/forms/output/BenefitsLineGraph.js:53]: This seems not to be working currently.

Note for future: May be worth abstracting some of the specific benefit chart frostings. @todo

File: src/localization/en.js [hunk 2]
```diff
@@ -122,12 +122,11 @@ export default {
...
-      selectBenefits_v1:   `Select the benefits you currently receive`,
-      hasSection8Label_v1: `Do you have Section 8 Housing?`,
-      hasSection8Hint_v1:  `Section 8 provides rental housing assistance.`,
-      hasSnapLabel_v1:     `Do you have SNAP?`,
-      hasSnapHint_v1:      `SNAP provides assistance with buying food`,
+      currentBenefits_v1:      `Current Benefits`,
+      selectBenefits_v1:       `Select the benefits you currently receive`,
+      has_section8_label_v1:   `Do you have Section 8 Housing?`,
+      has_snap_label_v1:       `Do you have SNAP?`,
+      has_xrayVision_label_v1: `Do you have X-ray vision?`,
```
Comment (knod) [inline src/localization/en.js]: lol! Maybe we should keep it in, though. I kind of like it. It'll be an easter egg for our fans :D

Comment (knod) [reply src/localization/en.js]: And it'll show up on the localization report! I love it!

**Review by knod — CHANGES_REQUESTED** [review]
Some awesome work here!

One note I couldn't add is that the `updateClientValue` needs to be changed too. The way it's setting shelter and housing stuff is weird and this messes it up. You'll see when you look at it. Let me know if you need another eye on it since it's a bit wacky. Generally searching the code for `hasSNAP` and `hasSection8` will probably reveal any stray problems.

Other than those two things, I _think_ we're good...

**Comment by knod** [comment]
Btw, that's the kind of problem I think figuring out #952 would help.

File: src/localization/en.js (hunk 2 above)
Comment (turnerhayes) [reply src/localization/en.js]: Haha, damn it, I added that to test arbitrary numbers of benefits. I thought I had removed it!

File: src/forms/output/BenefitsLineGraph.js (hunk 1 above)
Comment (turnerhayes) [reply src/forms/output/BenefitsLineGraph.js:53]: What effect is this supposed to have?

Comment (turnerhayes) [reply src/forms/output/BenefitsLineGraph.js:53]: Ah, nevermind, I see it

File: src/localization/en.js [hunk 3]
```diff
@@ -122,11 +122,10 @@ export default {
     },
 
     currentBenefits: {
-      currentBenefits_v1:      `Current Benefits`,
-      selectBenefits_v1:       `Select the benefits you currently receive`,
-      has_section8_label_v1:   `Do you have Section 8 Housing?`,
-      has_snap_label_v1:       `Do you have SNAP?`,
-      has_xrayVision_label_v1: `Do you have X-ray vision?`,
```
Comment (knod) [inline src/localization/en.js]: Dangit

**Review by knod — APPROVED** [review]
So because of my own past mistakes the benefit orders are wrong, but that's on me. The work in this PR is lookin' good to me 👍

**Pull request merged**